### ISSUE TITLE
Full example router.yaml or schema for validation/reference

### ISSUE THREAD

**Issue by hobbsh** [apollographql/router#1438]
**Describe the solution you'd like**
I have been trying to find a full example of `router.yaml` but I can't seem to. I think it would be useful to have a full example of it somewhere or at least a schema to reference. I'm assuming there is some sort of validation happening against a schema in the code but I could not find where that might be. Regardless, some sort of reference or schema that is updated along with the code would make configuration of the router a bit more straight-forward.

**Comment by o0Ignition0o** [comment]
Hi, and thanks for opening this issue!

The way I tend to deal with it, is by [adding configuration awareness to my editor](https://www.apollographql.com/docs/router/configuration/overview#configuration-awareness-in-your-text-editor).

TLDR: if you run `./router --schema` it will print the router's [JSON schema](https://json-schema.org/) configuration file.

This file can then be used as linter for your config.router.yaml file.

We went with this workflow for two reasons:
- The configuration file is pretty big already, and navigating through it can be hard
- The configuration file will change depending on the kind of plugin you have.

Importing your own router's schema configuration will always yield a schema configuration that is in sync with what you're running.

I hope it helps!

**Comment by abernix** [comment]
I think @o0Ignition0o's suggestion is probably the best bet and seems to answer your inquiry, so I'm going to close this.  Happy to re-open this if you feel it doesn't cover what you're looking for though!
